localstack/services/apigateway/provider.py

```diff
@@ -1192,6 +1198,121 @@ def filter_function(item):
+            raise BadRequestException(
+                "Model schema must have at least 1 property or array items defined"
+            )
+
+        model_id = short_uid()[:6]  # length 6 to make TF tests pass
+        model = Model(
+            id=model_id, name=name, contentType=content_type, description=description, schema=schema
+        )
+        store.rest_apis[rest_api_id].models[name] = model
+        remove_empty_attributes_from_model(model)
+        return model
+
+    def get_models(
+        self,
+        context: RequestContext,
+        rest_api_id: String,
+        position: String = None,
+        limit: NullableInteger = None,
+    ) -> Models:
+        store = get_apigateway_store(account_id=context.account_id, region=context.region)
+        if rest_api_id not in store.rest_apis:
+            raise NotFoundException(
+                f"Invalid API identifier specified {context.account_id}:{rest_api_id}"
+            )
+
+        models = [
+            remove_empty_attributes_from_model(model)
+            for model in store.rest_apis[rest_api_id].models.values()
+        ]
+        return Models(items=models)
+
+    def get_model(
+        self,
+        context: RequestContext,
+        rest_api_id: String,
+        model_name: String,
+        flatten: Boolean = None,
+    ) -> Model:
+        store = get_apigateway_store(account_id=context.account_id, region=context.region)
+        if rest_api_id not in store.rest_apis or not (
+            model := store.rest_apis[rest_api_id].models.get(model_name)
+        ):
+            raise NotFoundException(f"Invalid model name specified: {model_name}")
+
+        return model
+
+    def update_model(
+        self,
+        context: RequestContext,
+        rest_api_id: String,
+        model_name: String,
+        patch_operations: ListOfPatchOperation = None,
+    ) -> Model:
+        # manually update the model, not need for JSON patch, only 2 path supported with replace operation
+        # /schema
+        # /description
+        store = get_apigateway_store(account_id=context.account_id, region=context.region)
+        if rest_api_id not in store.rest_apis or not (
+            model := store.rest_apis[rest_api_id].models.get(model_name)
+        ):
+            raise NotFoundException(f"Invalid model name specified: {model_name}")
+
+        for operation in patch_operations:
+            path = operation.get("path")
+            if operation.get("op") != "replace":
+                raise BadRequestException(
+                    f"Invalid patch path  '{path}' specified for op 'add'. Please choose supported operations"
+                )
+            if path not in ("/schema", "/description"):
+                raise BadRequestException(
+                    f"Invalid patch path  '{path}' specified for op 'replace'. Must be one of: [/description, /schema]"
+                )
+
+            key = path.strip("/")
```

Comment:
nit: we are striping left and right slashes (`/`), but most likely the intent is just removing the leading slash, otherwise the check above `path in ("/schema" ...` would not work if it had a trailing slash.

Reply:
Right, yes, I'll just remove the first character of the string. Thanks! 👍